axum-macros/src/from_request.rs

```diff
@@ -0,0 +1,533 @@
+                    },
+                })
+            } else if peel_result_ok(&field.ty).is_some() {
+                Ok(quote_spanned! {ty_span=>
+                    #member: {
+                        ::axum::extract::FromRequest::from_request(req)
+                            .await
+                            .map(#into_inner)
+                    },
+                })
+            } else {
+                Ok(quote_spanned! {ty_span=>
+                    #member: {
+                        ::axum::extract::FromRequest::from_request(req)
+                            .await
+                            .map(#into_inner)
+                            .map_err(Self::Rejection::#rejection_variant_name)?
+                    },
+                })
+            }
+        })
+        .collect()
+}
+
+fn peel_option(ty: &syn::Type) -> Option<&syn::Type> {
+    let type_path = if let syn::Type::Path(type_path) = ty {
+        type_path
+    } else {
+        return None;
+    };
+
+    let segment = type_path.path.segments.last()?;
+
+    if segment.ident != "Option" {
+        return None;
+    }
+
+    let args = match &segment.arguments {
+        syn::PathArguments::AngleBracketed(args) => args,
+        syn::PathArguments::Parenthesized(_) | syn::PathArguments::None => return None,
+    };
+
+    let ty = if args.args.len() == 1 {
+        args.args.last().unwrap()
+    } else {
+        return None;
+    };
+
+    if let syn::GenericArgument::Type(ty) = ty {
+        Some(ty)
+    } else {
+        None
+    }
+}
+
+fn peel_result_ok(ty: &syn::Type) -> Option<&syn::Type> {
+    let type_path = if let syn::Type::Path(type_path) = ty {
+        type_path
+    } else {
+        return None;
+    };
+
+    let segment = type_path.path.segments.last()?;
+
+    if segment.ident != "Result" {
+        return None;
+    }
+
+    let args = match &segment.arguments {
+        syn::PathArguments::AngleBracketed(args) => args,
+        syn::PathArguments::Parenthesized(_) | syn::PathArguments::None => return None,
+    };
+
+    let ty = if args.args.len() == 2 {
+        args.args.first().unwrap()
+    } else {
+        return None;
+    };
+
+    if let syn::GenericArgument::Type(ty) = ty {
+        Some(ty)
+    } else {
+        None
+    }
+}
+
+fn extract_each_field_rejection(
+    ident: &syn::Ident,
+    fields: &syn::Fields,
+    vis: &syn::Visibility,
+) -> syn::Result<TokenStream> {
+    let rejection_ident = rejection_ident(ident);
+
+    let variants = fields
+        .iter()
+        .map(|field| {
+            let FromRequestAttrs { via } = parse_attrs(&field.attrs)?;
+
+            let field_ty = &field.ty;
+            let ty_span = field_ty.span();
+
+            let variant_name = rejection_variant_name(field)?;
+
+            let extractor_ty = if let Some((_, path)) = via {
+                if let Some(inner) = peel_option(field_ty) {
+                    quote_spanned! {ty_span=>
+                        Option<#path<#inner>>
```

Comment:
For consistency, if `peel_option` and `peel_result_ok` work with weird ways of writing `Option`, we shouldn't just assume it is in scope here right?

```suggestion
                        ::std::option::Option<#path<#inner>>
```